_includes/markdown/Typescript.md

```diff
@@ -0,0 +1,221 @@
+
+The [React TypeScript Cheatsheet](https://react-typescript-cheatsheet.netlify.app/) is a good resource to check how to type specific things in React (e.g Forms, Events etc).
+
+### Use function default arguments instead of `defaultProps`
+
+`defaultProps` has been deprecated in React 19, therefore prefer using function default arguments.
+
+```tsx
+type MyComponentProps = {
+    title?: string;
+};
+
+export const MyComponent = ({ title = 'Default Title' }: MyComponentsProps) => {
+    // ...
+}
+```
+
+Make sure the eslint rule `react/require-default-props` is set up to look for `defaultArguments` e.g `'react/require-default-props': ['error', { functions: 'defaultArguments' }]`;
+
+### Prefer explicitly declaring `children` instead of using `PropsWithChildren`
+
+`PropsWithChildren` will make `children` optional, but sometimes you do want to be very explicit and make it required. Therefore we generally recommend declaring `children` explicitly.
+
+```tsx
+type LayoutProps = {
+    children: React.Node;
+};
+
+// If Layout is not passed a children, TS will catch the error
+const Layout = (props: LayoutProps) {
+    return (
+        <main>
+            {children}
+        </main>
+    );
+}
+```
+
+### Avoid prop spreading
+
+Prop spreading is when you simply forward all props to another component e.g:
+
+```tsx
+// AVOID DOING THIS
+const PostProps = {
+    title: string,
+    category: string,
+    link: string
+    // other props
+};
+
+const Post = (props: PostProps) => {
+    return (
+        <article>
+            <PostContent {...props}>
+        </article>
+    );
+}
+```
+
+While there might be use cases where you need to use them, it's generally better to avoid them and pass props explicitly. In the example above it would be better to just pass a `post` object as a single prop.
+
+Prop spreading is especially dangerous when prop spreading into native DOM elements. So make sure to not ignore TypeScript errors and make sure your types are set up correctly so that TS can catch forwarded props incompatible with the target component/element.
+
+### Colocate types with React components
+
+For the most part, the React Component types should be colocated with the React component, meaning that it should be in the same file that the React component is written. Only hoist types to avoid circular dependencies and when you expect them to be reused/shared across many components/files.
+
+### Use `@types` packages if necessary
+
+If the library you're using does not ship types, check if there is type information available for that package in the [DefinatelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped) repo.
+
+### Global types definitions
+
+**Do not** put the application types in the global scope just to avoid importing them. You may only create global type definitions when needing to extend global types (e.g `window` object).
```

Comment:
```suggestion
**Do not** put the application types in the global scope just to avoid importing them. You may only create global type definitions when needing to extend global types (e.g. `window` object).
```